prompterator/models/openai_models.py

```diff
@@ -57,6 +58,25 @@ def __init__(self):
             self.client = AzureOpenAI(
                 api_version=api_version, azure_endpoint=endpoint, api_key=api_key
             )
+        elif self.openai_variant == "azure-default-credentials":
+            default_credential = DefaultAzureCredential()
+            token = default_credential.get_token(
+                "https://cognitiveservices.azure.com/.default")
```

Comment:
@dastrupl could we please extract this into an environment variable and add a bit of documentation to `README.md` about it? As you can easily imagine, people might ask for different scopes depending on their Azure tenants 🙂

Reply:
2 new commits added to address. Please let me know if it is Ok now.